File: python/stdlib/default/splitflow.py
try:
    from ...stdlib import Node as base
except ValueError:
    from stdlib import Node as base


class Node(base.Node):
    def __init__(self, verbose, args):
        super(Node, self).__init__("Splitflow 2", "default.splitflow", "", {"val": "Object"},
                                   {"left": "Object", "right": "Object"},
                                   "Splits the flow.", verbose)

    def tick(self, value):
        tags = {}
        if "tags" in value and "val" in value["tags"]:
            tags = {"left": value["tags"]["val"], "right": value["tags"]["val"]}
        return {"tags": tags, "left": value["val"], "right": value["val"]}

File: lua/stdlib/default/splitflow.lua
local myNode = {}

function myNode.init(node)
  node.tick = function(value)
    result = {}
    result.left = value.val
    result.right = value.val
    return result
  end
end

function myNode.spec(node)
  node.name = "Splitflow 2"
  node.inputs.val = "Object"
  node.outputs.left = "Object"
  node.outputs.right = "Object"
  node.desc = "Split the flow into two."
end

return myNode
